src/foam/nanos/ruler/RulerDAO.js

```diff
@@ -123,14 +172,93 @@
         }
       ],
       javaCode: `
-
       for ( Object key : sink.getGroupKeys() ) {
         List<Rule> group = ((ArraySink) sink.getGroups().get(key)).getArray();
         if ( ! group.isEmpty() ) {
+          Collections.sort(group, new foam.mlang.order.Desc(Rule.PRIORITY));
           new RuleEngine(x, this).execute(group, obj, oldObj);
         }
       }
       `
+    },
+    {
+      name: 'updateRules',
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        }
+      ],
+      javaCode: `DAO ruleDAO = ((DAO) x.get("ruleDAO")).where(EQ(Rule.DAO_KEY, getDaoKey()));
```

Comment:
Code is easier to read if you don't put large java methods in the model. Mark the class as abstract: true then move larger methods to the .java file. If abstract: true it will generate AbstractRulerDAO.java instead of RulerDAO.java. You then write RulerDAO that extends AbstractRulerDAO.

Reply:
easier to read from where exactly? You can always open generated file and read code from there?